src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -875,23 +876,28 @@ private void CheckNewModifier(Symbol symbol, bool isNew, DiagnosticBag diagnosti
                             {
                                 if (overridingProperty.GetMethod is object)
                                 {
+                                    MethodSymbol overriddenGetMethod = overriddenProperty.GetOwnOrInheritedGetMethod();
                                     checkValidNullableMethodOverride(
                                         overridingProperty.GetMethod.Locations[0],
-                                        overriddenProperty.GetMethod,
+                                        overriddenGetMethod,
                                         overridingProperty.GetMethod,
                                         diagnostics,
+                                        checkReturnType: true,
                                         // Don't check parameters on the getter if there is a setter
                                         // because they will be a subset of the setter
-                                        checkParameters: overridingProperty.SetMethod is null);
+                                        checkParameters: overridingProperty.SetMethod is null ||
+                                                         overriddenGetMethod?.AssociatedSymbol != overriddenProperty ||
+                                                         overriddenProperty.GetOwnOrInheritedSetMethod()?.AssociatedSymbol != overriddenProperty);
```

Comment:
Could consider extracting this check to a helper, as we use it in multiple places.

Reply:
> Could consider extracting this check to a helper, as we use it in multiple places.

I do not believe it is worth doing just for portion of the condition, the entire condition is different.

---
In reply to: [296387584](https://github.com/dotnet/roslyn/pull/36663#discussion_r296387584) [](ancestors = 296387584)